extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -104,23 +106,40 @@ export class AmpUserNotification extends AMP.BaseElement {
     this.elementId_ = assert(this.element.id,
         'amp-user-notification should have an id.');
 
-    assert(this.element.hasAttribute('data-show-if-href'),
-        `"amp-user-notification" (${this.elementId_}) ` +
-        'should have "data-show-if-href" attribute.');
-    /** @private @const {string} */
-    this.showIfHref_ = assertHttpsUrl(
-        this.element.getAttribute('data-show-if-href'), this.element);
+    /** @const @private {!Promise<!Storage>} */
+    this.storagePromise_ = storageFor(this.win_);
 
-    assert(this.element.hasAttribute('data-dismiss-href'),
-        `"amp-user-notification" (${this.elementId_}) ` +
-        'should have "data-dismiss-href" attribute.');
+    /** @private @const {?string} */
+    this.storageKey_ = isExperimentOn(this.getWin(), 'amp-storage') ?
+        'amp-user-notification:' + this.elementId_ : null;
 
-    /** @private @const {string} */
-    this.dismissHref_ = assertHttpsUrl(
-        this.element.getAttribute('data-dismiss-href'), this.element);
+    /** @private @const {?string} */
+    this.showIfHref_ = this.element.getAttribute('data-show-if-href');
+    if (this.showIfHref_) {
+      assertHttpsUrl(this.showIfHref_, this.element);
+    }
 
-    this.userNotificationManager_
-        .registerUserNotification(this.elementId_, this);
+    /** @private @const {?string} */
+    this.dismissHref_ = this.element.getAttribute('data-dismiss-href');
+    if (this.dismissHref_) {
+      assertHttpsUrl(this.dismissHref_, this.element);
+    }
+
+    // TODO(dvoytenko): document storage behavior in these methods once
+    // experiment is removed.
+    if (!this.storageKey_) {
+      assert(this.showIfHref_,
+          `"amp-user-notification" (${this.elementId_}) ` +
+          'should have "data-show-if-href" attribute.');
+      assert(this.dismissHref_,
+          `"amp-user-notification" (${this.elementId_}) ` +
+          'should have "data-dismiss-href" attribute.');
+    }
+
+    if (!this.win_.AMP_TEST) {
```

Comment:
Removed. Found another way to do this. But ultimately, I want unit-tests to be untainted by service integration - it's tested separately.